src/client_backend/dynamic_grpc/dynamic_grpc_client.h

```diff
@@ -161,32 +154,20 @@ class DynamicGrpcClient {
   /// by gRPC when sending requests. By default compression is not used.
   /// \return Error object indicating success or failure of the request.
   Error StartStream(
-      OnCompleteFn callback, bool enable_stats = true,
+      OnCompleteFn callback = [](InferResult*) {}, bool enable_stats = true,
       const Headers& headers = Headers(),
       grpc_compression_algorithm compression_algorithm = GRPC_COMPRESS_NONE);
 
   /// Stops an active grpc bi-directional stream, if one available.
   /// \return Error object indicating success or failure of the request.
   Error StopStream();
 
-  /// Returns the inference statistics of the client.
-  const InferStat& ClientInferStat() { return infer_stat_; }
-
- protected:
-  // Update the infer stat with the given timer
-  Error UpdateInferStat(const tc::RequestTimers& timer);
-  // Enables verbose operation in the client.
-  bool verbose_;
-
-  // The inference statistic of the current client
-  InferStat infer_stat_;
-
- private:
   // Generic bi-directional stream using dynamic protobuf message.
   std::unique_ptr<grpc::GenericClientAsyncReaderWriter> bidi_stream_;
   std::unique_ptr<grpc::ClientContext> grpc_context_;
   std::unique_ptr<grpc::CompletionQueue> completion_queue_;
   bool stream_started_{false};
+  bool writesdone_called_{false};
```

Comment:
Could this variable be renamed to something more semantic?

What does writes done being called actually mean?

Does that mean a single inference has had all of its messages sent?

Maybe a name like `inference_sent_` or `inference_messages_sent_` or something like that might be more educative?

Reply:
Sure, calling [WritesDone()](https://grpc.github.io/grpc/cpp/classgrpc_1_1_client_reader_writer_interface.html#a52f4e5d5ac7fe0e4995cb337aa0ecfc8) means half-closing the stream from the client side, so maybe `stream_half_closed_`? This name will make it clear why we are restarting the stream in the beginning of the inference.

Reply:
I think incorporating the concept of inference in this variable name would help a new reader understand its role within the context of the entire class, whose purpose is to send inference requests (messages).

Reply:
Idk, maybe it's fine to leave it as it is. It's just that when I read "writes done called" it doesn't help me understand what this variable is gonna be used for.

Reply:
I hear your concern. I think I'm just trying to keep the grpc client to contain logics that makes sense in the context of grpc communication. I'll change to `stream_half_closed_` and make a small comment that WriteDone closes the stream. How's that?

Reply:
Sounds good 😄